internal/server/data/data.go

```diff
@@ -194,16 +194,20 @@ func (t *Transaction) GormDB() *gorm.DB {
 // Rollback the transaction. If the transaction was already committed then do
 // nothing.
 func (t *Transaction) Rollback() error {
-	if t.committed.Load() {
+	if t.completed.Load() {
 		return nil
 	}
-	return t.DB.Rollback().Error
+	err := t.DB.Rollback().Error
+	if err == nil {
+		t.completed.Store(true)
+	}
```

Comment:
Added this so that calling rollback multiple times does not error. This allows a request handler to end a transaction early to keep transaction lifetime short.